src/style/style_layer.js

```diff
@@ -210,7 +209,7 @@ class StyleLayer extends Evented {
             'paint': this._transitionablePaint && this._transitionablePaint.serialize()
         };
 
-        if (this.visibility === 'none') {
+        if (this.visibility) {
```

Comment:
This will always set `output.layout.visibility` to `'none'` regardless of the value of `this.visibility`

Reply:
I think I’ve fixed this in a subsequent commit. Or I misunderstand. There’s a test that ensures it can be set to “visible”.

Reply:
You're right, I was viewing a single commit instead of all the changes.